### DISCUSSION TITLE
Composable Navigation which holds Composables for back stack

### DISCUSSION THREAD

**Discussion by simonvar** [arkivanov/Decompose#400] [Q&A]
Hello!

I need to implement stack navigation with Jetpack Compose, but i have patially transparent screens. 
So i want to implement something like Children() composable, which never destroys composable at back stack.

Any suggestions for better way of implementing this?

**Comment by arkivanov** [comment]
The easiest way would be just display all screens in a Box, in their natural order, wrapping each screen with `key(it.configuration) { ... }`. Would this work?

**Reply by simonvar** [reply]
Basically – yes, but i still want use animation of adding/removing screen.

**Reply by arkivanov** [reply]
You can try wrapping each screen with `AnimatedVisibility`. Would this work?

**Reply by simonvar** [reply]
🤔 I think this should work. 
Thanks for suggestion, i will try it later.

**Reply by simonvar** [reply]
@arkivanov AnimatedVisibility working for animation of adding, but not for removing. Removing configuration from stack just removes whole composable without animation.
I know that is not scope of decompose, but maybe you know method of animation for decomposition?

**Reply by arkivanov** [reply]
Yeah, I missed that point! You can try following this nice guide of choosing a right animation - https://developer.android.com/jetpack/compose/animation

E.g. maybe AnimatedContent could work, I didn't try though. Let me know hot it goes.
